pkgs/stdenv/generic/setup.sh

```diff
@@ -121,14 +121,17 @@ exitHandler() {
     set +e
 
     if [ -n "${showBuildStats:-}" ]; then
-        times > "$NIX_BUILD_TOP/.times"
-        local -a times=($(cat "$NIX_BUILD_TOP/.times"))
+        buildTimes=()
+        while read -r line; do
```

Comment:
I wonder if this could work to give a bit more natural code ordering
```suggestion
        times | while read -r line; do
```

Reply:
Yeah I tried that first, I didn't know as well, but there was a shellcheck warning, it taught me something.
Basically if you do a pipe in a loop it doesn't modify the variable in the outer context. (I'm sure you've figured it out, just putting it out for other people wondering).